sdk/modelsrepository/Azure.IoT.ModelsRepository/src/MetadataScheduler.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.IoT.ModelsRepository
+{
+    /// <summary>
+    /// MetadataScheduler tracks when metadata should be fetched from a repository.
+    /// </summary>
+    internal class MetadataScheduler
+    {
+        private DateTime _lastFetched;
+        private readonly TimeSpan _desiredElapsedTimeSpan;
+        private bool _initialFetch;
+
+        /// <param name="desiredElapsedTimeSpan">A desired <c>TimeSpan</c> that is used to calculate elapsed time periods.</param>
+        public MetadataScheduler(TimeSpan desiredElapsedTimeSpan)
+        {
+            _lastFetched = DateTime.MinValue;
+            _desiredElapsedTimeSpan = desiredElapsedTimeSpan;
+            _initialFetch = true;
+        }
+
+        /// <summary>
+        /// Updates the last fetched DateTime attribute of the scheduler to <c>DateTime.Now</c>.
```

Comment:
Should it be `DateTime.UtcNow`? same on line 35

Reply:
We just need a relative DateTime with respect to the system clock to compare with. There isn't data being exchanged with other systems/endpoints, its relative to the client instance. So I think either DateTime.Now or DateTime.UtcNow would work. Is there a case you are thinking where DateTime.Now doesn't work?

Reply:
I figured if we decided to add some SDK logs to the operations and make it a bit more chatty, we could use UTC times to make it consistent and easier to work with.

Reply:
Ah OK, I gotchu

Reply:
Changed to `DateTime.UtcNow`.